Assets/Runtime/Conditions/AOECondition.cs

```diff
@@ -45,11 +45,34 @@ public override void OnApplied(ConditionBank bank)
                     {
                         foreach (var thisCond in bank.Conditions)
                         {
-                            targetBank.Apply(thisCond);
+                            float duration = thisCond.Duration;
+                            Condition newCondition;
+                            // Case uses pattern casting!
+                            switch (thisCond)
```

Comment:
Very good job for figuring out how to use the switch statement with pattern matching.

Just a small DRY (Don't Repeat Yourself) optimization: you are calling `SetTimeAndApply()` in each of your switch cases. The only thing that's changing in your switch cases is the condition variable. That means you can instead move `SetTimeAndApply()` outside of the switch statement. So it would look more like this:

```
foreach (var thisCond in bank.Conditions)
{
    targetBank.Apply(thisCond);
    float duration = thisCond.Duration;
    Condition newCondition;
    // Case uses pattern casting!
    switch (thisCond)
    {
        case SlowCondition condition:
        {
            newCondition = new SlowCondition(condition.slowPercent);
            break;
        }
        case DOTCondition condition:
        {
            newCondition = new DOTCondition(condition.Damage);
            break;
        }
    }

    SetTimeAndApply(newCondition, condition, targetBank);
}

```

The C# developers realized this is a common thing and gave us a handy little shorthand way of dealing in this situation:

```
foreach (var thisCond in bank.Conditions)
{
    targetBank.Apply(thisCond);
    float duration = thisCond.Duration;
    
    Condition newCondition = (thisCondition) switch {
        SlowCondition condition => new SlowCondition(condition.slowPercent),
        DOTCondition condition => new DOTCondition(condition.Damage),
    };

    SetTimeAndApply(newCondition, condition, targetBank);
}
```

Reply:
There's also a good chance that this switch pattern thing will become a problem if we introduce lots of conditions, but we'll cross that bridge when we come to it.

Reply:
I had a reason for this, based on an error that basically said what you said, where if a type wasn't caught, it would try to pass a non-initialized condition down. Rider provided the elegant solution of setting it to null before, and now I use:
```c#
if (newCondition != null) {targetBank.Apply(newCondition)}
``` 
So now, if it changes, it gets caught! And if it doesn't change, it doesn't apply. Now, a bunch of null conditions don't start showing up on enemies.